pkg/dataplane/server/dataplane.go

```diff
@@ -40,6 +41,8 @@ type Dataplane struct {
 	clusters           map[string]*cluster.Cluster
 	listeners          map[string]*listener.Listener
 	listenerEnd        map[string]chan bool
+	clustersLock       sync.Mutex
+	listenerLock       sync.Mutex
```

Comment:
I don't think this is needed as I don't see any concurrent access to the listeners map.

Reply:
Listener is also added and removed similar to clusters.